src/coreclr/jit/codegenxarch.cpp

```diff
@@ -2825,7 +2834,15 @@ void CodeGen::genCodeForInitBlkUnroll(GenTreeBlk* node)
                 regSize = 8;
             }
 #endif
-            if (bytesWritten + regSize > size)
+            if (bytesWritten + regSize > size && bytesWritten < size)
```

Comment:
It appears that `bytesWritten < size` will always be true here, since that's the condition of the `while` loop above. That means we always do the "shiftBack" logic. But it also means the `else if (bytesWritten + regSize > size)` condition and it's `break` are dead code.